keps/sig-scheduling/3280-guarantee-pdb-when-preemption-happens/README.md

```diff
@@ -0,0 +1,962 @@
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+If a user sets a PodDisruptionBudget (PDB) for some low-priority pods and sets the `PreemptionPolicy` 
+to `RequiredNotPreempted`, high-priority pods will not be able to violate the PDB and preempt 
+these pods during the scheduling process. This may result in high-priority pods being unable 
+to schedule while low-priority pods continue to run normally, which may make troubleshooting 
+more complex for the user. To address this, additional logging or event descriptions will be 
+added in the implementation to clearly inform the user of the reason why preemption did not occur.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+A new field named `PreemptionPolicy` will be added to `PodDisruptionBudgetSpec`.
+In order to address the issue mentioned above, a new field `PreemptionPolicy` will 
+be added to the `PodDisruptionBudgetSpec`. This field will indicate whether or not 
+violation of the PDB will be allowed during the scheduler preemption process. The 
+default value for this field will be `PreferNotPreempted`, which is consistent with 
+current behavior. Users will be able to set this field to determine whether or not 
+PDB violation will be allowed during scheduler preemption. Additionally, in order 
+to avoid system stability issues, even if the victim's PDB is not allowed to be 
+violated during the preemption process, the preempter with priority ClassName of 
+`system-cluster-critical` or `system-node-critical` will still be able to violate 
+the victim's PDB during preemption.
+
+```go
+// PodDisruptionBudgetSpec is a description of a PodDisruptionBudget.
+type PodDisruptionBudgetSpec struct {
+	MinAvailable *intstr.IntOrString
+	Selector *metav1.LabelSelector
+	MaxUnavailable *intstr.IntOrString
+    
+  // PreemptionPolicy indicates whether pdb can be violated 
+  // during preemption. Pods with priority ClassName of
+  // system-cluster-critical or system-node-critical 
+  // can violate the victim's pdb during preemption.
+  // Defaults to PreferNotPreempted if unset.
+  // +optional
+  PreemptionPolicy *PreemptionPolicy
```

Comment:
Maybe we can be a bit more explicit:

```yaml
permitPreemptionBy: SystemCritical|HigherPriority
```

cc @liggitt as SIG Auth TL and API reviewer